lib/Transforms/InstCombine/InstCombineCompares.cpp

```diff
@@ -2058,6 +2058,11 @@ Instruction *InstCombiner::visitICmpInstWithInstAndIntCst(ICmpInst &ICI,
     // smaller constant, which will be target friendly.
     unsigned Amt = ShAmt->getLimitedValue(TypeBits-1);
     if (LHSI->hasOneUse() &&
+        // @LOCALMOD-BEGIN
+        // We don't want to introduce non-power-of-two integer sizes for PNaCl's
+        // stable wire format, so modify this transformation for NaCl.
+        isPowerOf2_32(TypeBits - Amt) && (TypeBits - Amt) >= 8 &&
```

Comment:
It looks like code from our pnacl legacy, avoiding weird nonstandard integers (which we don't know how to legalize). I'm not sure of more details, but to be safe, you could condition it on the target being emscripten.